pkg/controller/atlasdeployment/advanced_deployment.go

```diff
@@ -77,12 +79,91 @@ func (r *AtlasDeploymentReconciler) ensureAdvancedDeploymentState(ctx *workflow.
 	}
 }
 
+func handleSearchNodes(ctx *workflow.Context, deployment *akov2.AtlasDeployment, projectID string) workflow.Result {
+	// Get the current state from Atlas
+	ctx.Log.Debug("starting search node processing")
+	defer ctx.Log.Debug("finished search node processing")
+
+	// NOTE: Despite being an array/slice, the Atlas API only allows 1 item.
+	nodesInAtlasEmpty := false
+	currentNodesInAtlas, httpResp, err := ctx.SdkClient.AtlasSearchApi.GetAtlasSearchDeployment(ctx.Context, projectID, deployment.GetDeploymentName()).Execute()
+	if err != nil {
+		if httpResp == nil {
+			ctx.Log.Debugf("unable to get current search nodes status: %v", err)
+			return workflow.Terminate(workflow.SearchNodesReady, err.Error())
+		}
+
+		switch httpResp.StatusCode {
+		// TODO: Should be only NotFound: CLOUDP-239015
+		case http.StatusBadRequest, http.StatusNotFound:
+			ctx.Log.Debug("no search nodes in atlas found")
+			nodesInAtlasEmpty = true
+		default:
+			return workflow.Terminate(workflow.SearchNodesReady, err.Error())
+		}
+	}
+
+	nodesInAkoEmpty := deployment.Spec.DeploymentSpec.SearchNodes == nil || len(deployment.Spec.DeploymentSpec.SearchNodes) == 0
```

Comment:
This can be reduced to:
```suggestion
	nodesInAkoEmpty := len(deployment.Spec.DeploymentSpec.SearchNodes) == 0
```
because it will also catch the `nil` case.